### PR TITLE
Fix chooser buttons focus color is too dark in dark mode

### LINKED ISSUE #10875: Chooser buttons focus colour is too dark in dark mode
### Is your proposal related to a problem?

Chooser buttons become dull after being visited leading editors to think the button is disabled. 

To replicate, click a chooser, dismiss the modal.

Before:

image

After:

image

### Describe the solution you'd like

CSS has the following rule:

```css
.button.chooser__choose-button {
    align-items: center;
    border-color: #0000;
    color: var(--w-color-text-label);
    color: var(--w-color-text-button-outline-default);
    display: flex;
    font-size: .875rem;
    font-weight: 500;
    line-height: 1.3;
    padding: .375rem;
}
```

Amend the selector to include the visited state:

```css
.button.chooser__choose-button, .button.chooser__choose-button:visited {
    align-items: center;
    border-color: #0000;
    color: var(--w-color-text-label);
    color: var(--w-color-text-button-outline-default);
    display: flex;
    font-size: .875rem;
    font-weight: 500;
    line-height: 1.3;
    padding: .375rem;
}
```

Buttons retain original colouring on tests.
Tested with image, page and generic choosers.

### PR DESCRIPTION
**Pull request by rohitsrma**
Fixes #10875

**Before**

Screenshot (41)

**After (Updated in 2nd commit)**

Screenshot (42)

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Review by lb- — CHANGES_REQUESTED** [review]
Great start @rohitsrma  - but a bit more is needed here to get this over the line.

See my comment inline.

File: client/scss/components/_chooser.scss [hunk 1]
```diff
@@ -28,7 +28,7 @@ $preview-size: 2.625rem; // 42px
 
   &:hover,
   &:focus {
-    color: theme('colors.surface-button-hover');
+    color: theme('colors.text-button-outline-default');
```
[the remark this answers is not in the document]
Comment (rohitsrma) [reply client/scss/components/_chooser.scss:31]: Thanks, @lb- -, for the detailed suggestions. I initially thought we needed the same color for hover/focus with an outline, as shown in the `proposed hover state` in the issue. I apologize for the misunderstanding.

So, to use `teal`, do we first need to add it in https://github.com/wagtail/wagtail/blob/b7beb36df6bc89e6152eae6c7d2dabaee7bd660a/client/src/tokens/colors.js
 
I can't see it already defined in the file. Please correct me if I am wrong.

[the remark this answers is not in the document]
Comment (lb-) [reply client/scss/components/_chooser.scss:31]: Teal is the secondary colour. I think we have a secondary-400.

https://github.com/wagtail/wagtail/blob/b7beb36df6bc89e6152eae6c7d2dabaee7bd660a/client/src/tokens/colors.js#L139

[the remark this answers is not in the document]
Comment (rohitsrma) [reply client/scss/components/_chooser.scss:31]: Sorry for not clarifying earlier. When I mentioned `teal` I was referring to `teal-200 and 100`. I think it is not currently defined in the file.

[the remark this answers is not in the document]
Comment (lb-) [reply client/scss/components/_chooser.scss:31]: Ahh yep. Good call - these shades will probably be needed yes. Thanks for checking.

**Comment by rohitsrma** [comment]
Thanks @lb- , I have made some changes. Could you please take a look?

**Review by lb- — APPROVED** [review]
@rohitsrma thank you for this.

It looks like https://github.com/wagtail/wagtail/pull/11463 from @thibaudcolas has come in on top and actually resolved a few of these added variables.

Even then though, this still does not fix the problem as the chooser buttons are either the exactly same colour (after #11463) on hover or in your PR still to dark.

For the sake of a solution I will push up a change to your PR that just makes the choosre hover/focus colour use `color: theme('colors.text-link-hover');`

This makes it the same value on light mode (`var(--w-color-secondary-400)`) but means we can use a might lighter colour in dark mode. `var(--w-color-secondary-75)`.

@thibaudcolas  please feel free to do another update if you think this is the wrong way to go.

## Before (with changes from #11463 )

| Dark | Light |
|-|-|
| Screenshot 2024-01-23 at 12 09 49 pm | Screenshot 2024-01-23 at 12 09 33 pm |

## After

| Dark | Light (no change) |
|-|-|
| Screenshot 2024-01-23 at 12 07 51 pm | Screenshot 2024-01-23 at 12 08 08 pm |

**Pull request merged**